app/models/concerns/porter_utils.rb

```diff
@@ -0,0 +1,53 @@
+require "active_support/concern"
+
+module PorterUtils
```

Comment:
Chunks of this were copied from `PorterEncoder`.  But I reorganized the primary methods to reflect the structure of a [porter job](https://github.com/PRX/Porter#input-message-format).

Which has an ID, Source, Tasks, and Callbacks.

The common things (ID, Source, Callbacks) are defined here.  But the Tasks are up to the specific `Task` sub-model.

Reply:
reading a bit through the Porter readme to better understand Tasks and Callbacks. i think a lot of my questions have been stemming from unfamiliarity with Porter and the workflow process between our apps and Porter.